language-testutils/src/test/java/de/jplag/testutils/datacollector/TokenPositionTestData.java

```diff
@@ -0,0 +1,92 @@
+package de.jplag.testutils.datacollector;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import de.jplag.Language;
+import de.jplag.ParsingException;
+import de.jplag.Token;
+import de.jplag.testutils.TmpFileHolder;
+import de.jplag.util.FileUtils;
+
+/**
+ * Test sources with token information Reads token position test specifications form a file and provides the token
+ * information for tests. The sources cna be used as regular test sources.
+ */
+public class TokenPositionTestData implements TestData {
+    private final List<String> sourceLines;
+    private final List<TokenData> expectedTokens;
+
+    private final String descriptor;
+
+    /**
+     * @param testFile The file containing the test specifications
+     * @throws IOException If the file cannot be read
+     */
+    public TokenPositionTestData(File testFile) throws IOException {
+        this.sourceLines = new ArrayList<>();
+        this.expectedTokens = new ArrayList<>();
+        this.descriptor = "(Token position file: " + testFile.getName() + ")";
+        this.readFile(testFile);
+    }
+
+    private void readFile(File testFile) throws IOException {
+        List<String> testFileLines = FileUtils.readFileContent(testFile).lines().toList();
+        int currentLine = 0;
+
+        for (String sourceLine : testFileLines) {
+            if (sourceLine.charAt(0) == '>') {
+                this.sourceLines.add(sourceLine.substring(1));
+                currentLine++;
+            }
+
+            if (sourceLine.charAt(0) == '$') {
+                int col = sourceLine.indexOf('|');
```

Comment:
`column`. Generally, there are a lot of unnecessary abbreviations throughout the code; try to avoid them.